pkg/distributor/push.go

```diff
@@ -169,7 +173,8 @@ func handler(
 			if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
 				code, msg = int(resp.Code), string(resp.Body)
 			} else {
-				code, msg = toHTTPStatus(ctx, err, limits), err.Error()
+				_, code = toGRPCHTTPStatus(ctx, err, limits)
```

Comment:
You might want the grpc error code for the logline below.

Reply:
this is not used by normal mimir handler, i would just leave it as right now.